app/models/post.rb

```diff
@@ -1,3 +1,13 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments
+  has_many :likes
+
+  def update_user_post_counter
+    author.update(posts_count: author.posts.count)
+  end
+
```

Comment:
[OPTIONAL] It is a good practice to use rails callbacks in scenarios like this. It helps keep your code DRY.